test/wallaby/browser_test.exs

```diff
@@ -1,7 +1,122 @@
 defmodule Wallaby.BrowserTest do
-  use ExUnit.Case, async: true
+  use ExUnit.Case
+
+  import Wallaby.SettingsTestHelpers
 
   alias Wallaby.Browser
+  alias Wallaby.Session
+
+  defmodule TestDriver do
+
+    def start_link() do
+      Agent.start_link(fn -> [] end, name: __MODULE__)
+    end
+
+    def visit(%Session{} = session, path) do
+      Agent.update(__MODULE__, fn (visits) ->
+        [path | visits]
+      end)
+      session
+    end
+
+    def assert_visited(url) do
+      unless Enum.member?(visits(), url) do
+        raise ExUnit.AssertionError,
+          """
+          #{inspect url} was not visited.
+
+          Visited urls:
+          #{visits() |> Enum.map(&"  #{inspect &1}") |> Enum.join("\n")}
+          """
+      end
+    end
+
+    defp visits do
+      Agent.get(__MODULE__, fn (visits) -> visits end)
+    end
+  end
+
+  describe "visit/2" do
+    setup do
+      ensure_setting_is_reset(:wallaby, :base_url)
+      {:ok, _} = TestDriver.start_link()
+      :ok
+    end
+
+    test "relative path without leading slash, base url with trailing slash" do
+      Application.put_env(:wallaby, :base_url, "http://example.com/")
+
+      session = session_for_driver(TestDriver)
+      Browser.visit(session, "form.html")
+
+      TestDriver.assert_visited("http://example.com/form.html")
```

Comment:
I like putting the assertion in the driver, no implementation leakage that way! :+1: